### PR TITLE
Refactor cleanup jobs for forward-progress alerting

### PR DESCRIPTION
**Pull request by sethvargo**
This does not actually change the alerts yet, just refactors the existing code to make it easier to export the new metrics and remove many unused metrics to simplify the code flow.

**Release Note**

```release-note
Refactor background cleanup jobs for forward-progress alerting. This removes unused metrics about the number of exposures/files deleted and simplifies the logic handling.
```

### CODE REVIEW

File: terraform/vars.tf [hunk 1]
```diff
@@ -195,7 +195,7 @@ variable "cleanup_exposure_worker_cron_schedule" {
 
 variable "cleanup_export_worker_cron_schedule" {
   type    = string
-  default = "0 */6 * * *"
+  default = "0 */4 * * *"
```
Comment (sethvargo) [inline terraform/vars.tf:198]: @mikehelmick PTAL here

**Review by whaught — DISMISSED** [review]
/lgtm
/hold for @mikehelmick to take a look

File: internal/cleanup/cleanup.go [hunk 2]
```diff
@@ -57,54 +58,52 @@ type exposureCleanupHandler struct {
...
+
+		if count, err := h.database.DeleteStatsBefore(ctx, cutoff); err != nil {
+			merr = multierror.Append(merr, fmt.Errorf("failed to delete stats: %w", err))
+		} else {
+			logger.Infow("purged statistics", "count", count)
+		}
+	}()
+
+	if merr != nil {
+		logger.Errorw("failed to cleanup exposures", "errors", merr.WrappedErrors())
```
Comment (mikehelmick) [inline internal/cleanup/cleanup.go:100]: log message needs updating, it's not just exposures now

File: internal/cleanup/cleanup.go [hunk 3]
```diff
@@ -133,49 +132,75 @@ type exportCleanupHandler struct {
...
+
+		if count, err := h.database.DeleteFilesBefore(ctx, cutoff, h.blobstore); err != nil {
+			merr = multierror.Append(merr, fmt.Errorf("failed to delete files: %w", err))
+		} else {
+			logger.Infow("purged files", "count", count)
+		}
+	}()
+
+	if merr != nil {
+		logger.Errorw("failed to cleanup exposures", "errors", merr.WrappedErrors())
```
Comment (mikehelmick) [inline internal/cleanup/cleanup.go:162]: this is export files, not exposures, right?

File: terraform/vars.tf (hunk 1 above)
Comment (mikehelmick) [reply terraform/vars.tf:198]: this is fine.

File: internal/cleanup/cleanup.go (hunk 2 above)
Comment (sethvargo) [reply internal/cleanup/cleanup.go:100]: The underlying `errors` are still wrapped with the context of whether it was exposures or stats that failed

File: internal/cleanup/cleanup.go (hunk 3 above)
Comment (sethvargo) [reply internal/cleanup/cleanup.go:162]: Yea

**Review by mikehelmick — APPROVED** [review]

**Pull request merged**